examples/canvas_more/canvas_more/app.py

```diff
@@ -0,0 +1,216 @@
+from contextlib import contextmanager
+from math import pi, cos, sin, atan
+
+import toga
+from toga.style import Pack
+from toga.style.pack import COLUMN
+from travertino.size import at_least
+
+
+class ExampleCanvasApp(toga.App):
+    def startup(self):
+        # Initial width and height for the Canvas
+        iwidth = 600
+        iheight = 400
+
+        # Set up main window
+        self.main_window = toga.MainWindow(title=self.name)
+
+        # Setup the style selection box
+        self._style_select = toga.Selection(
+            items=list(self.styles), style=Pack(height=32), on_select=self.set_style
+        )
+        # Setup the canvas
+        self._canvas = toga.Canvas(style=Pack(flex=1), on_resize=self.resize_contents)
+        self._canvas.intrinsic = \
+            Pack.IntrinsicSize(width=at_least(iwidth), height=at_least(iheight))
+
+        # Add the content on the main window
+        box = toga.Box(
+            style=Pack(direction=COLUMN),
+            children=[self._style_select, self._canvas]
+        )
+        self.main_window.content = box
+
+        # Setup canvas transformation objects use to handle window moving and
+        # resizing so that actual drawing code can use a fixed range of
+        # coordinates
+        self._canvas_translate = self._canvas.translate(iwidth/2, iheight/2)
```

Comment:
If you need the initial size of the *window*, you could use `self.main_window.size`... admittedly it would be preferable to use the size of the *widget*, but give the current layout and capabilities of toga, that's about as close as you're going to get.

Reply:
Yeah, since I have another widget on the window, the windows size does not really give me what I need. Also this might not be accurate as well if someone is using a window manager that automatically maximizes the window.

I'd rather keep things as they are until we get around to implementing some of the better ways to handle this that we've discussed elsewhere already.

Reply:
Again - I *do* mind, which is why I raised it as a review comment. I'm not willing to introduce an example to the codebase that encourages the use of private APIs or less than ideal programming practices. I'd rather see a *less* featured example, or a fix for the underlying problem.